### DISCUSSION TITLE
Using this in a GPO

### DISCUSSION THREAD

**Discussion by chesh1r3k** [N-able/ScriptsAndAutomationPolicies#22] [Ideas]
If you use this script in a GPO to install the agent, what will happen if you have a multi-site setup? Let's say the agent gets installed in Site A with a customer number of 101 and then the user moves to Site B with a customer number of 102 and a GPO to run this script for each site respectively.  Will the agent re-install with the new site information? If that happens, will it leave behind a zombie agent in Site A?  I'm used to running Tim Wiser's GPO and his VBS script had protections in place from creating zombie agents.  I see this script really only deploys the agent and doesn't really do any checks to see if the agent was previously installed and running in another site. 

Also, may I suggest that you add [System.Net.ServicePointManager]::ServerCertificateValidationCallback = {$true} before $bindingURL for people that have self signed certificates so the script doesn't fail.

**Comment by Robby-Swartenbroekx** [comment]
I think you are refering to the DeployTheNCAgent
This is a default template script provided by Solarwinds to show us how to get the token from the API.
if you use this as is in a GPO, you need to leave your JWT token on the DC's of your customer. With this JWT, they could access the API completely (or to the parts where that account has access to). So don't think this is intented to use as is in a GPO.

Also Tim and Ryan created a far more complex script to be used in GPO's. Don't think we'll receive something like that from SolarWinds.

**Reply by chesh1r3k** [reply]
Yes, sorry, I didn't realize when starting a discussion, it didn't tie directly to what script I was looking at.  And, that was my second thought on the DeployNCAgent script.  I'm guessing this was just a proof of concept for accessing the API as I couldn't see any other real use for this script as if you're signing on a new customer, it would be just as easy to have the customer specific agent on a thumb drive sneaker netting between computers instead of downloading the agent each time with the script.
